core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -1192,11 +1195,69 @@ object Chunk {
 
     override def apply(n: Int): A = vector(n)
 
-    override def foreach(f: A => Unit): Unit = vector.foreach(f)
+    override def foreach(f: A => Any): Unit = vector.foreach(f)
 
     override def toArray[A1 >: A](n: Int, dest: Array[A1]): Unit = { val _ = vector.copyToArray(dest, n, length) }
   }
 
+  final class BitChunk private (private val bytes: Chunk[Byte]) extends NonEmpty[Boolean] { self =>
+
+    def apply(n: Int): Boolean =
+      (bytes(n >> 3) & (1 << (7 - (n & 7)))) != 0
+
+    def foreach(f: Boolean => Any): Unit = {
+      var i = 0
+      while (i < length) {
```

Comment:
Eventually this can be optimized so you iterate over the bytes in `bytes` and call `f` 8 times for each byte.